Mesh template time series report related stuff (#19)
Volue-Public/energy-smp-docs: docs/timeseries-reports/reports_intro.md

# Mesh template reports

A time series report in Mesh context is based on a template definition. The Mesh model objects are the data providers.

The report definition uses Mesh model queries to find time series to include in an activated report presented to the user through a client user interface.

A report definition is currently stored in Oracle database (table RC_SET ++) and is edited by the SmP Calculator tool. The definition format is basically the same as for SmP time series reports that was introduced in the beginning of 90's. A report definition consists of the following sections:

- General report attribute-value pairs
- Column definitions with attribute-value pairs (also referred to as report items)

On each column definition there must be a reference to something that can provide time series data, a time series source.  
In general, the following source types are available:

- Time series name or time series key (attribute DBI_KEY)
- Local time series expression (a block wrapped with #SLV_START ... ##SLV_END)
- Time series query (attribute TS_BINDING)

> In Mesh template reports it is only the **TS_BINDING** approach that is supported.  

Reviewer: ```suggestion
Mesh template reports only supports the **TS_BINDING** approach.
```

The column definition attribute named `TS_BINDING` is used to look up time series to be part of an activated report.
Some of the collected time series may not immediately appear in the visible view onto the report due to page configurations and hierarchical grouping.
There are also options to dynamically include more time series by retrieving series that are used to provide initial series through calculations.
Adding previous versions of initial series is another option to dynamically extend the report.

The `TS_BINDING` attribute **value** in Mesh context is associated
with Mesh model instances that hold time series. To describe where to look for time series we use the Mesh model *definition*.
The TS_BINDING value is a query and not a fixed reference, therefore a template report will adapt to the current state
of the Mesh model.

A query against a given Mesh model may result in zero, one or *many* time series. This dynamic behavior also has
impact on how other report attributes are defined, for instance by using macros that make the
value of attributes adapt to the concrete time series found. `$OBJ_NAME` is an example of such a macro and will expand to the

> name of the Mesh model object that owns the concrete time series attribute.

Reviewer: ```suggestion
A query against a given Mesh model may result in zero, one or *many* time series. This dynamic behaviour also impact how other report attributes are defined, for instance by using macros that make the value of attributes adapt to the specific time series found. `$OBJ_NAME` is an example of such a macro and will expand to the name of the Mesh model object that owns the time series attribute.
```

## Template report activation

A template report definition is based on a Mesh model *definition* but needs additional context to be activated:

- A Mesh model where there are instances of the object types found in the Mesh model *definition*
- A start-point definition

> - A time period that defines which part of each time series that are read/calculated and visualized in the client that activated the report.

Reviewer: ```suggestion
- A time period that defines which part of each time series that are read/calculated and visualised in the client that activated the report.
```
